specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2/backuprestore.json

```diff
@@ -193,6 +193,64 @@
           }
         }
       }
+    },
+    "/restore/key/{keyName}": {
+      "put": {
+        "tags": [
+          "SelectiveKeyRestore"
+        ],
+        "operationId": "SelectiveKeyRestoreOperation",
+        "description": "Restores all key versions of a given key using user supplied SAS token pointing to a previously stored Azure Blob storage backup folder",
+        "parameters": [
+          {
+            "name": "keyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the key to be restored from the user supplied backup"
+          },
+          {
+            "name": "restoreBlobDetails",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/RestoreOperationParameters"
+            },
+            "description": "The Azure blob SAS token pointing to a folder where the previous successful full backup was stored"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Started selective key restore operation from the previously stored backup",
+            "headers": {
+              "Retry-After": {
+                "description": "The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.",
+                "type": "integer"
+              },
+              "Azure-AsyncOperation": {
```

Comment:
Given this, should the `RestoreOperation` model and these related methods be LROs? We'd probably define them as such in the client SDKs, and marking them as so - if they abide by the LRO guidelines for Azure services - would generate the right code for those languages using generation.

Reply:
A single key restore is implemented as 'POST /keys/keyname/restore'. Selective restore is exactly that except the key material is coming from Full HSM backup, rather than a single key backup file. So I would have expected selective restore to use 'POST /keys/keyname/restore' and the request body will have different parameters distinguishing whether the source is single key file backup or full hsm backup stored in Storage.

Reply:
If that's a problem for standard LROs, we can still do it in our client SDK and should. We did that for delete and recover of keys, etc.